cumulus/parachains/integration-tests/emulated/common/src/macros.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_export]
+macro_rules! test_parachain_is_trusted_teleporter {
```

Comment:
This should be a simple `fn` generic on `<SENDER_PARA, RECEIVER_PARA>` with `assets` and `amount` as simple params.

Macros suck for readability (and confuse IDEs) and I would definitely avoid when unnecessary (like here IIUC).

Reply:
I got that comment before about the macro. I was forced to use macro here becase unless I am missing something it is not possible to constrain the trait impl type to have specific enum variants.
The method depends for example on `RuntimeEvent`, which is an enum... and the compiler can't tell if `PolkadotXcm`, `XcmpQueue`, etc exist for every `Chain`.